controllers/druid/handler.go

```diff
@@ -391,6 +395,158 @@ func checkCrashStatus(sdk client.Client, m *v1alpha1.Druid) {
+
+	// In case size dont match, patch the pvc with the desiredsize from druid CR
+	if desVolumeClaimTemplateSize.String() != pvcSize.String() {
+		for i := range pvc {
+			patch := client.MergeFrom(pvc[i].DeepCopy())
+			pvc[i].Spec.Resources.Requests[v1.ResourceStorage] = desVolumeClaimTemplateSize
+			err := sdk.Patch(context.TODO(), pvc[i], patch)
+			if err != nil {
+				e := fmt.Errorf("Failed to Patch [PVC:%s] due to [%s]", pvc[i].Name, err.Error())
+				logger.Error(e, e.Error(), "name", drd.Name, "namespace", drd.Namespace)
+				sendEvent(sdk, drd, v1.EventTypeWarning, "DELETE_FAIL", e.Error())
+				return
+			} else {
+				msg := fmt.Sprintf("[PVC:%s] successfully Patched with [Size:%s]", pvc[i].Name, pvcSize.String())
+				logger.Info(msg, "name", drd.Name, "namespace", drd.Namespace)
+				sendEvent(sdk, drd, v1.EventTypeNormal, "PATCH_SUCCESS", msg)
+			}
+		}
+	}
+	// after deleting the sts, it takes some time for the sts to come
+	time.Sleep(10 * time.Second)
+	return
+}
+
+// Create pvcList, reference sts using component label
+func getPVCList(sdk client.Client, nodeSpec *v1alpha1.DruidNodeSpec, drd *v1alpha1.Druid, nodeSpecUniqueStr string) []*v1.PersistentVolumeClaim {
+
+	pvcList := makePersistentVolumeClaimListEmptyObj()
+	listOpts := []client.ListOption{
+		client.InNamespace(drd.Namespace),
+		client.MatchingLabels(map[string]string{
+			"nodeSpecUniqueStr": nodeSpecUniqueStr,
+		}),
+	}
+
+	if err := sdk.List(context.TODO(), pvcList, listOpts...); err != nil {
+		e := fmt.Errorf("failed to list pvc for [%s:%s] due to [%s]", drd.Kind, drd.Name, err.Error())
+		sendEvent(sdk, drd, v1.EventTypeWarning, "LIST_FAIL", e.Error())
+		logger.Error(e, e.Error(), "name", drd.Name, "namespace", drd.Namespace)
+		return nil
+	}
+
+	// create a slice []*pvc
+	pvc := make([]*v1.PersistentVolumeClaim, 0)
+	for i := range pvcList.Items {
+		pvc = append(pvc, &pvcList.Items[i])
+	}
+
+	return pvc
+
+}
+
+// desVolumeClaimTemplateSize: the druid CR holds this value for a sts volumeclaimtemplate
+// currVolumeClaimTemplateSize: the sts owned by druid CR holds this value in volumeclaimtemplate
+// pvcSize: the pvc referenced by the sts holds this value
+// type of vars is resource.Quantity. ref: https://godoc.org/k8s.io/apimachinery/pkg/api/resource
+func getVolumeClaimTemplateSizes(sts *appsv1.StatefulSet, nodeSpec *v1alpha1.DruidNodeSpec, pvc []*v1.PersistentVolumeClaim) (desVolumeClaimTemplateSize, currVolumeClaimTemplateSize, pvcSize resource.Quantity) {
+
+	for i := range nodeSpec.VolumeClaimTemplates {
+		desVolumeClaimTemplateSize = nodeSpec.VolumeClaimTemplates[i].Spec.Resources.Requests[v1.ResourceStorage]
+	}
+
+	for i := range sts.Spec.VolumeClaimTemplates {
+		currVolumeClaimTemplateSize = sts.Spec.VolumeClaimTemplates[i].Spec.Resources.Requests[v1.ResourceStorage]
+	}
+
+	for i := range pvc {
+		pvcSize = pvc[i].Spec.Resources.Requests[v1.ResourceStorage]
+	}
+
+	return desVolumeClaimTemplateSize, currVolumeClaimTemplateSize, pvcSize
```

Comment:
> it seems you are assuming that there is only one pvc in the nodeSpec.VolumeClaimTemplates array ?
yes


```nodeSpec.VolumeClaimTemplates``` is a single pvc, since our goal here specifically to increase the size of sts referenced pvc's ie those pvc, in case other pvc's are mounted on sts they wont be specified in volumeClaimTemplates. Those will be mounted on volumeMounts and Volumes specifying pvc claims there. something like this. 
```
volumes:
- name: mysql-data
  persistentVolumeClaim:
    claimName: mysql-data-pod-name-0.  
```

Reply:
`nodeSpec.VolumeClaimTemplates` is an array and user can totally specify two/more pvc templates in there.